tests/serialization/test_flows.py

```diff
@@ -134,6 +135,43 @@ def test_serialize_container_environment():
     assert deserialized.storage.registry_url == storage.registry_url
 
 
+def test_serialize_result_handler():
+    serialized = FlowSchema().dump(Flow(name="n", result_handler=JSONResultHandler()))
+    assert serialized["result_handler"]["type"] == "JSONResultHandler"
+    deserialized = FlowSchema().load(serialized)
+    assert deserialized.result_handler
```

Comment:
These asserts don't demonstrate that the deserialization worked and restored the specified result handler